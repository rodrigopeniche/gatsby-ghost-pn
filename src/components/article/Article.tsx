import * as React from "react"
import "./article-body.css"
import "./special.css"
import * as ReactMarkdown from "react-markdown"
import ArticleContainer from "./ArticleContainer"
import Sidebar from "../sidebar/Sidebar"
import { navigate } from "gatsby"

const Article = ({
    data,
    missionStatement,
}: {
    data: any
    missionStatement?: boolean
}) => {
    if (data == undefined || data["sections"] == undefined || data["sections"] == null) {
        navigate("/404")
        return (<></>)
    }
    return (
        <>
            {"sidebar" in data ? <Sidebar text={data["sidebar"]} /> : null}
            <ArticleContainer missionStatement={missionStatement}>
                {data["sections"].map(
                    (
                        {
                            name,
                            content,
                            cells,
                        }: {
                            name: string
                            content: string | null
                            cells: { content: string }[] | null
                        },
                        i: string | number
                    ) => (
                        <React.Fragment key={i}>
                            {content && (
                                <ReactMarkdown
                                    className={name}
                                    source={content}
                                    escapeHtml={false}
                                />
                            )}
                            {cells && (
                                <div className={name}>
                                    {cells.map(({ content }, i) => (
                                        <ReactMarkdown
                                            key={i}
                                            className={"cell-" + i}
                                            source={content}
                                            escapeHtml={false}
                                        />
                                    ))}
                                </div>
                            )}
                        </React.Fragment>
                    )
                )}
            </ArticleContainer>
        </>
    )
}

export default Article
