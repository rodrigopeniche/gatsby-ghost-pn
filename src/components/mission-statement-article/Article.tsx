import * as React from "react"
import "./article-body.css"
import "./special.css"
import * as ReactMarkdown from "react-markdown"
import ArticleContainer from "./ArticleContainer"

const Article = ({ data }: { data: any }) => {
    return (
        <ArticleContainer>
            {data.map(
                (
                    {
                        name,
                        content,
                        cells,
                    }: {
                        name: string
                        content: string | null
                        cells: { content: string }[] | null
                    },
                    i: string | number
                ) => (
                    <React.Fragment key={i}>
                        {content && (
                            <ReactMarkdown
                                className={name}
                                source={content}
                                escapeHtml={false}
                            />
                        )}
                        {cells && (
                            <div className={name}>
                                {cells.map(({ content }, i) => (
                                    <ReactMarkdown
                                        key={i}
                                        className={"cell-" + i}
                                        source={content}
                                        escapeHtml={false}
                                    />
                                ))}
                            </div>
                        )}
                    </React.Fragment>
                )
            )}
        </ArticleContainer>
    )
}

export default Article
