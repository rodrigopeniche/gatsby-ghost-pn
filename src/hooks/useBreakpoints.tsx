import * as React from "react"
import { useState, useEffect, createContext, useContext } from "react"

const defaultValue = {}

const BreakpointContext = createContext(defaultValue)

const BreakpointProvider = ({
    children,
    queries,
}: {
    children: React.ReactNode | React.ReactNode[]
    queries: { [key: string]: string }
}) => {
    const [queryMatch, setQueryMatch] = useState({})

    useEffect(() => {
        const mediaQueryLists: { [key: string]: any } = {}
        const keys = Object.keys(queries)
        let isAttached = false

        const handleQueryListener = () => {
            const updatedMatches = keys.reduce(
                (acc: { [key: string]: any }, media) => {
                    acc[media] = !!(
                        mediaQueryLists[media] && mediaQueryLists[media].matches
                    )
                    return acc
                },
                {}
            )
            setQueryMatch(updatedMatches)
        }

        if (window && window.matchMedia) {
            const matches: { [key: string]: any } = {}
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media] = window.matchMedia(queries[media])
                    matches[media] = mediaQueryLists[media].matches
                } else {
                    matches[media] = false
                }
            })
            setQueryMatch(matches)
            isAttached = true
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media].addListener(handleQueryListener)
                }
            })
        }

        return () => {
            if (isAttached) {
                keys.forEach(media => {
                    if (typeof queries[media] === "string") {
                        mediaQueryLists[media].removeListener(
                            handleQueryListener
                        )
                    }
                })
            }
        }
    }, [queries])

    return (
        <BreakpointContext.Provider value={queryMatch}>
            {children}
        </BreakpointContext.Provider>
    )
}

function useBreakpoint(): { [key: string]: boolean } {
    const context = useContext(BreakpointContext)
    if (context === defaultValue) {
        throw new Error("useBreakpoint must be used within BreakpointProvider")
    }
    return context
}
export { useBreakpoint, BreakpointProvider }
