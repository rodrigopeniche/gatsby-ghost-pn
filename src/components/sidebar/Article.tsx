import * as React from "react"
import articleLogo from "./images/articleLogo.png"
import { useBreakpoint } from "../../hooks/useBreakpoints"

const Article = () => {
    const { md } = useBreakpoint()

    const articleStyles: React.CSSProperties = {
        marginRight: "15px",
        minWidth: "97%",
        alignContent: "center",
        ...(md && {
            minWidth: "100%",
            marginTop: "0",
        }),
    }

    return <img src={articleLogo} style={articleStyles}></img>
}

export default Article
