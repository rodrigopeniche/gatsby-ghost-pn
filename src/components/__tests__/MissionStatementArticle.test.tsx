import * as React from "react"
import * as renderer from "react-test-renderer"
import { BreakpointProvider } from "../../hooks/useBreakpoints"
import Article from "../mission-statement-article/Article"

describe("Mission statement article", () => {
    it("renders correctly", () => {
        const children = [{ name: "children", content: "<div>children</div>" }]
        const tree = renderer.create(
            <BreakpointProvider queries={{}}>
                <Article data={children} />
            </BreakpointProvider>
        )
        expect(tree.toJSON()).toMatchSnapshot()
        expect(tree.root.findByType("span").props).toEqual({
            dangerouslySetInnerHTML: { __html: "<div>children</div>" },
        })
    })
})
