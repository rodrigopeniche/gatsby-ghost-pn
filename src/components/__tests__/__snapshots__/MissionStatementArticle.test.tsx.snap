// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission statement article renders correctly 1`] = `
<div
  style={
    Object {
      "boxSizing": "border-box",
      "marginLeft": "11px",
      "marginRight": "11px",
      "maxWidth": "929px",
      "minWidth": "250px",
    }
  }
>
  <div
    className="children"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div>children</div>",
        }
      }
    />
  </div>
</div>
`;
